include/deal.II/cgal/surface_mesh.h

```diff
@@ -66,6 +78,50 @@ namespace CGALWrappers
     const typename dealii::Triangulation<dim, spacedim>::cell_iterator &cell,
     const dealii::Mapping<dim, spacedim> &                              mapping,
     CGAL::Surface_mesh<CGALPointType> &                                 mesh);
+
+  /**
+   * Performs corefinement and boolean operations on two deal.II cells and put
+   * the result in a CGAL::Surface_mesh object.
+   *
+   * Extensive examples can be found in the CGAL documentation at
+   * https://doc.cgal.org/latest/Polygon_mesh_processing/index.html#coref_coref_subsec
+   *
+   * @tparam CGALPointType
+   * @tparam dim
+   * @tparam spacedim
+   * @param[in] cell1
+   * @param[in] cell2
+   * @param[out] outsm
+   * @param[in] boolean_operation BooleanOperation::intersection_op and
+   * BooleanOperation::union_op perform intersection and union of cells,
+   * respectively. If not specified, this parameter is defaulted to
+   * BooleanOperation::simple_corefinement, and only a corefinement is performed
+   */
+  template <typename CGALPointType, int dim, int spacedim>
+  void
+  corefine_and_compute_boolean_operation_dealii_cells(
+    const typename Triangulation<dim, spacedim>::cell_iterator &cell1,
+    const typename Triangulation<dim, spacedim>::cell_iterator &cell2,
+    CGAL::Surface_mesh<CGALPointType> &                         outsm,
+    const BooleanOperation &boolean_operation =
+      BooleanOperation::only_corefinement);
+
+  /**
+   * Build a Quadrature<spacedim> formula to integrate over the volumetric
+   * region described by a CGAL::Surface_mesh, by filling the interior with
+   * simplices and collecting points and weights to initegrate over it.
```

Comment:
What happens if the surface mesh is not closed?

Reply:
I cannot think to a case where having a non-closed surface makes sense, after all we're intersecting deal.II cells in that function.

Reply:
I know, but does the function check that surface mesh is closed or not? You should document what happens in that case, and if the function throws an exception, you should state this here.